feat: support dynamic table of contents

NozomuIkuta opened the pull request:
<!--- Provide a general summary of your changes in the title above -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)


## Description
<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
<!--- If it resolves an open issue, please link to the issue here. For example "Resolves: #137" -->

Resolves: #575. 

I added an option `content.markdown.tocDepth`, by which users can define which heading are included in table of contents. For example, if it is set as `3` (default), `h2` and `h3` are included in table of contents.

There are 3 special cases to handle:

First case is where `content.markdown.tocDepth` is set as `1`. Currently, `h1` seems to be treated as the title of the article rather than content. So, I let the module log an info message.

Second case is where `content.markdown.tocDepth` is set as `0`. I handle this case in same way as above.

Third case is where `content.markdown.tocDepth` is set as more than `6`. Since deepest heading tag is `h6`, I let the module logs an message.

By the way, the reason why I don't design the option as originally described in the issue is that writing every tag names looks redundant to me.

## Checklist:
<!--- Put an `x` in all the boxes that apply. -->
<!--- If your change requires a documentation PR, please link it appropriately -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes (if not applicable, please state why)

NozomuIkuta commented on docs/content/en/configuration.md:
```diff
@@ -213,6 +213,14 @@ export default {
+
+- Type: `Number`
+- Default: `3`
+- Version: **>= v1.11.0**
```
This is just an placeholder, and supposed to be set correctly later.

NozomuIkuta commented on docs/content/en/configuration.md:
```diff
@@ -213,6 +213,14 @@ export default {
+- Default: `3`
+- Version: **>= v1.11.0**
+
+Define until which heading levels is included in table of contents.
```
I'm not sure this sentence is valid in English...

benjamincanac replied:
I would say: `Maximum heading depth to include in the table of contents` (source: https://github.com/remarkjs/remark-toc#optionsmaxdepth)

NozomuIkuta commented on packages/content/lib/utils.js:
```diff
@@ -10,6 +10,7 @@ const getDefaults = ({ dev = false } = {}) => ({
   fullTextSearchFields: ['title', 'description', 'slug', 'text'],
   nestedProperties: [],
   markdown: {
+    tocDepth: 3,
```
For backward compatibility, I set default value of `tocDepth` as 3, so that `h2` and `h3` are included in table of contents.

NozomuIkuta reviewed (commented):
I added some inline comments.

benjamincanac reviewed (approved):
LGTM 🙂

NozomuIkuta commented on packages/theme-docs/src/components/app/AppToc.vue:
```diff
@@ -25,7 +25,10 @@
               class="block text-sm scrollactive-item transition-padding ease-in-out duration-300 hover:pl-1"
```
`hover:pl-1` might result in "too dynamic" animation  for deeper headings.

NozomuIkuta commented:
@benjamincanac 

Thank you for reviewing the draft PR!

I modified parts of code and updated the PR message.
Please review it again when you have time.

NozomuIkuta commented:
Hmm, it seems tests that I added take too long...they passed in my local, though.
I will divide then into `describe()` , so that I can set up longer timeout as like the other tests.
Or, do you have any other ideas?

NozomuIkuta commented:
OK, now all the tests are passed by splitting tests for the new feature. 🙌

benjamincanac commented:
Great work @NozomuIkuta, thank you very much 🙂 I think it's ready to be merged!
